pkg/controller/account/account_controller.go

```diff
@@ -427,29 +412,38 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 				reqLogger.Info("Account pending AWS limits verification")
 			}
 		}
-		err = r.Client.Status().Update(context.TODO(), currentAcctInstance)
-		if err != nil {
-			return reconcile.Result{}, err
-		}
-
 	}
 
 	// If Account CR has `stats.rotateCredentials: true` we'll rotate the temporary credentials
 	// the secretWatcher is what updates this status field by comparing the STS credentials secret `creationTimestamp`
+	credentialsRotated := false
 	if accountNeedsCredentialsRotated(currentAcctInstance) {
 		reqLogger.Info(fmt.Sprintf("rotating CLI credentials for %s", currentAcctInstance.Name))
 		err = r.RotateCredentials(reqLogger, awsSetupClient, currentAcctInstance)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
+		credentialsRotated = true
 	}
 	if accountNeedsConsoleCredentialsRotated(currentAcctInstance) {
 		reqLogger.Info(fmt.Sprintf("rotating console URL credentials for %s", currentAcctInstance.Name))
 		err = r.RotateConsoleCredentials(reqLogger, awsSetupClient, currentAcctInstance)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
+		credentialsRotated = true
 	}
+	if credentialsRotated {
+		// Set `status.RotateCredentials` to false now that they have been updated
+		currentAcctInstance.Status.RotateCredentials = false
```

Comment:
I like that we're consolidating the actual push of the status update, but this isn't clearing the `RotateConsoleCredentials` status. You would want to leave the local status updates (L438, and its counterpart for `RotateConsoleCredentials = false`) inside the conditions above, and make this condition _only_ push the `Status().Update()`. 

Personally, I think it's goodness if `Rotate[Console]Credentials` is still responsible for updating the account status locally. Can we consider changing `Rotate[Console]Credentials` to returna a bool indicating whether it did that or not?